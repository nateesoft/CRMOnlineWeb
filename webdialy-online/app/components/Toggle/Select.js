import styled from 'styled-components';

const Select = styled.select`
  line-height: 1em;
  background-color: transparent;
  border-style: none;
  margin-right: 10px;
  border: 1px solid white;
`;

export default Select;
