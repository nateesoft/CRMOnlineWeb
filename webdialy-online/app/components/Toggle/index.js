/**
 *
 * LocaleToggle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Select from './Select';
import ToggleOption from '../ToggleOption';

const useStyles = makeStyles(() => ({
  textWhite: {
    color: 'white',
  },
}));
function Toggle(props) {
  const classes = useStyles();
  let content = <option>--</option>;

  if (props.values) {
    content = props.values.map(value => (
      <ToggleOption key={value} value={value} message={props.messages[value]} />
    ));
  }

  return (
    <Select value={props.value} onChange={props.onToggle} className={classes.textWhite}>
      {content}
    </Select>
  );
}

Toggle.propTypes = {
  onToggle: PropTypes.func,
  values: PropTypes.array,
  value: PropTypes.string,
  messages: PropTypes.object,
};

export default Toggle;
