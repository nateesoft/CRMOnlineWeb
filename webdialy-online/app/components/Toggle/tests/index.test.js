import React from 'react';
import renderer from 'react-test-renderer';

import Toggle from '../index';

describe('Toggle components', () => {
  it('test snapshot', () => {
    const component = renderer.create(<Toggle />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
