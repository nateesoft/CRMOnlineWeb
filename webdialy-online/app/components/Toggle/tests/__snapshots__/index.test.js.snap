// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle components test snapshot 1`] = `
<select
  className="makeStyles-textWhite-1 Select-qnzjgx-0 cQQWoM"
>
  <option>
    --
  </option>
</select>
`;
